packages/at_persistence_secondary_server/lib/src/log/commitlog/commit_log_keystore.dart

```diff
@@ -68,6 +68,15 @@ class CommitLogKeyStore
         commitEntry!.commitId = internalKey;
         // update entry with commitId
         await _getBox().put(internalKey, commitEntry);
+        // Delete old commit entry for the same key from the commit log
+        // For now, just delete the previous entry from _commitLogCacheMap
+        // Eventually, after the compaction kicks in all of the old entries would be deleted.
+        // Subsequently, we can then disable the compaction job.
+        // When commitLogCacheMap.keys.length == getBox().length then we have fully compacted it
+        if (_commitLogCacheMap[commitEntry.atKey!] != null) {
+          await _getBox()
+              .delete(_commitLogCacheMap[commitEntry.atKey!]?.commitId);
```

Comment:
For completeness, please add `&& _commitLogCacheMap[commitEntry.atKey!]?.commitId != null` to the if condition

Reply:
Updated the IF condition as above.